proof/invariant-abstract/DetSchedInvs_AI.thy

```diff
@@ -2291,17 +2291,17 @@ abbreviation valid_ready_qs :: "'z state \<Rightarrow> bool" where
 
 lemmas valid_ready_qs_def = valid_ready_qs_2_def valid_ready_queued_thread_2_def
 
-definition bounded_release_time_2 :: "time \<Rightarrow> time \<Rightarrow> bool" where
-  "bounded_release_time_2 curtime reltime \<equiv>
-     unat reltime \<le> unat curtime + unat kernelWCET_ticks + unat MAX_PERIOD"
+definition bounded_release_time_2 :: "time \<Rightarrow> bool" where
+  "bounded_release_time_2 reltime \<equiv>
+     unat reltime + 2 * unat MAX_PERIOD \<le> unat max_time"
```

Comment:
If it's not too much work, can we hide this `2` under a definition or (input) abbreviation?

Reply:
This `2` doesn't appear anywhere else, so I don't think we'd gain anything from having this as a definition or abbreviation. Happy to make a definition or abbreviation if others would like it, though

Reply:
Cool, could you add a short comment, just a little snippet so we know why it's 2 and not 1 or 3? (Presumably it's because we the overrun code could push an sc `MAX_PERIOD` into the future, and the span of the refills could fill out another `MAX_PERIOD`?)

Reply:
Howsabout:

"This predicate is currently used only in `refill_update_valid_refills`, in particular, to show that performing `refill_update` will not result in overflow. The constant `2` is required because `refill_update` will schedule a refill at most `MAX_PERIOD` from the head `r_time`, of `r_amount` at most `MAX_PERIOD`"

Reply:
Love it.